torch/multiprocessing/spawn.py

```diff
@@ -126,9 +130,29 @@ def join(self, timeout=None):
             return len(self.sentinels) == 0
 
         # Assume failure. Terminate processes that are still alive.
+        # Try SIGTERM then SIGKILL if the process isn't going down.
+        # The reason is related to python signal handling explained
+        # below. We have seen processes getting stuck not handling
+        # SIGTERM.
+        # https://fb.workplace.com/groups/1543508939745889/permalink/1548605689236214/
```

Comment:
nit: better to write out the explanation, as oss users can't read the workplace link.  (and its a pretty self-contained explanation anyway)